src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java

```diff
@@ -202,6 +204,13 @@
                 return new String[]{"No default architecture for tensor type:" + tensorType.name()};
             }
         }
+
+        IntelGKLUtils utils = new IntelGKLUtils();
+        if (utils.isAvxSupported() == false)
```

Comment:
Can you throw a `UserException` from the tool's `onStartup()` method instead of doing the check here in `customCommandLineValidation()`, or is there a reason why you're doing the check here?

The error message should also be more helpful (eg., it should contain a pointer to the latest version of Tensorflow that runs on non-AVX hardware).

Reply:
Also, AVX, not AVX2 is the minimum hardware required.

Reply:
@droazen @lucidtronix Can't over-ride onStartup() because the base class [VariantWalker]( gatk/src/main/java/org/broadinstitute/hellbender/engine/VariantWalker.java ) made the method `final` - the advice in the comment for that method suggests placing "startup" code in `onTraversalStart`.  Is `customCommandLineValidation` perhaps the "least bad" place to put the check?


Another problem: adding the advice to users on how to access an older/compatible TensorFlow build looks like it could be problematic too: how is our AVX check to know whether the installed build of TensorFlow requires AVX?  We could check the version - but a user might try a non-AVX-requiring user-created build (@mepowers linked to a v1.8 TensorFlow that doesn't require it, for example).

Reply:
@EdwardDixon Sorry, I meant `onTraversalStart()`, not `onStartup()`. Precondition checks like this belong in `onTraversalStart()` -- `customCommandLineValidation()` is for injecting hooks into the arg parser.

As for the second issue, I'd suggest that we add a `--disable-avx-check` argument to the tool. The `UserException` message can then say something like this:  "*This tool requires AVX instruction set support by default due to its dependency on recent versions of the TensorFlow library. If you have an older (pre-1.6) version of TensorFlow installed that does not require AVX, you may attempt to re-run the tool with the --disable-avx-check argument to bypass this check. Note that such configurations are not officially supported.*"